src/modules/member/services/member.service.ts

```diff
@@ -99,6 +107,83 @@ export class MemberService {
     });
   }
 
+  /**
+   * member 와 majorSkill 을 매핑
+   */
+  async mappingMajorSkill(
+    member: MemberEntity,
+    params: CreateMemberMajorSkillMappingRequestParamDto,
+  ): Promise<{ count: number }> {
+    // 현재 유저가 가진 major 와 다른 major 에 접근하려는 경우
+    if (member.majorId !== params.majorId) {
+      throw new ForbiddenException(
+        '유저의 major 와 다른 major 는 접근이 불가능합니다.',
+      );
+    }
+
+    // 모든 majorSkill 이 존재하는지
+    // major 에 속하는 majorSkill 인지 확인하기 위해 값을 가져온다.
+    const majorSkills: MajorSkillEntity[] =
+      await this.prismaService.majorSkill.findMany({
+        where: {
+          id: {
+            in: params.majorSkillIds,
+          },
+        },
+      });
+
+    // majorSkill 이 모두 존재하지 않는다면 에러
+    if (majorSkills.length !== params.majorSkillIds.length) {
+      throw new NotFoundException('존재하지 않는 majorSkill 이 존재합니다.');
+    }
+
+    // 모든 majorSkill 이 현재 major 의 하위 집합인지 확인
+    const isMajorSkill: boolean = majorSkills.every((majorSkill) => {
+      return majorSkill.majorId === params.majorId;
+    });
+
+    // major 에 속하지 않은 majorSkill 이라면 에러
+    if (!isMajorSkill) {
+      throw new BadRequestException(
+        'majorSkill 중 major 에 속하지 않은 skill 이 존재합니다.',
+      );
+    }
+
+    // 현재 유저랑 mapping 돼있는 majorSkill 을 mapping 하는 경우를 체크하기 위해 값을 뽑아온다.
+    const exMemberMajorSkillMapping: MemberMajorMappingEntity | null =
+      await this.prismaService.memberMajorSkillMapping.findFirst({
+        where: {
+          memberId: params.id,
+          majorSkillId: {
+            in: params.majorSkillIds,
+          },
+        },
+      });
+
+    // 이미 mapping 된 관계를 만드려는 경우 에러
+    if (exMemberMajorSkillMapping) {
+      throw new BadRequestException(
+        '이미 존재하는 member 의 majorSkill 이 존재합니다.',
+      );
+    }
+
+    // bulk insert 를 위해 전처리
+    const toCreateMemberMajorSkillMapping: Pick<
+      MemberMajorMappingEntity,
+      'majorSkillId' | 'memberId'
+    >[] = params.majorSkillIds.map((majorSkillId) => {
+      return {
+        majorSkillId,
+        memberId: params.id,
+      };
+    });
+
```

Comment:
return 없이 구현 불가능인가요?

Reply:
```suggestion
    // bulk insert 를 위해 전처리
    const toCreateMemberMajorSkillMapping: Pick<
      MemberMajorMappingEntity,
      'majorSkillId' | 'memberId'
    >[] = params.majorSkillIds.map((majorSkillId) => ({
        majorSkillId,
        memberId: params.id,
      })
    });
```
얘를 리턴없이 구현하게되면 아마 이런 형태가 돼서 저는 줄바꿈이 생기거나 객체를 리턴해줘야하면 중괄호를 사용해서 return 을 명시하는 편인데 return 을 빼는게 나을까요???

Reply:
저 또한 줄바꿈이 생기게 되면 return을 넣어주는 편입니다